### ISSUE TITLE
Fix Request: HttpClientBuilderWithoutPoolSize false positive

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#286]
Following example:
```
public class PoolingHttpClientConnectionManagerFactory extends AbstractFactoryBean<PoolingHttpClientConnectionManager> {

    private Registry<ConnectionSocketFactory> registry;
    private ConnectionProperties connectionProperties;
    private PoolingHttpClientConnectionManager poolingHttpClientConnectionManager;

    public PoolingHttpClientConnectionManagerFactory(Registry<ConnectionSocketFactory> registry, ConnectionProperties connectionProperties) {
        this.registry = registry;
        this.connectionProperties = connectionProperties;
    }

    @Override
    public Class<?> getObjectType() {
        return PoolingHttpClientConnectionManager.class; // flagged but okay
    }
[...]
}
```

**Comment by jborgers** [comment]
No false positive seen in the test case anymore in pmd7 version.
